src/installer/pkg/sfx/Microsoft.NETCore.App/ReadyToRun.targets

```diff
@@ -16,5 +31,34 @@
                     JitPath="@(_clrjit)"
                     DiaSymReader="$(_diaSymReaderPathIfExists)" />
     </ItemGroup>
+    <ItemGroup>
+      <Crossgen2Tool Include="$(Crossgen2Exe)" JitPath="$(JitLibrary)" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Hack to patch DOTNET_ROOT for the duration of Crossgen2 compilation to unblock -->
+  <!-- Crossgen2 bring-up before SDK 6.0 Preview 2 propagates to the runtime repo. -->
+  <!-- https://github.com/dotnet/runtime/issues/48252 -->
+
+  <PropertyGroup>
+    <OriginalDotnetRootValue />
+    <DOTNET_ROOT />
+  </PropertyGroup>
+
+  <Target Name="PatchDotnetRootBeforeRunningCrossgen2" BeforeTargets="_CreateR2RImages">
+    <PropertyGroup>
+      <OriginalDotnetRootValue>$(DOTNET_ROOT)</OriginalDotnetRootValue>
+      <DOTNET_ROOT>$(RepoRoot)</DOTNET_ROOT>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="RestoreDotnetRootAfterRunningCrossgen2" AfterTargets="_CreateR2RImages">
+    <PropertyGroup>
+      <DOTNET_ROOT>$(OriginalDotnetRootValue)</DOTNET_ROOT>
```

Comment:
The only problem with this is it will eventually change this property, some time after _CreateR2RImages ran. The only 2 ways I know for somewhat strict ordering are using a dependent target (the you don't own case, but there's an extensibility model, and it's cleaner), or the CallTarget route (which has few good uses, and unless we own _CreateR2RImages, it won't be a good idea..

Reply:
The exact problem here is that, while we "own" _CreateR2RImages, we do so in a different repo (sdk) and the entire trick here is about not having to wait for its propagation to the runtime repo as that will take about two months or more. If I could merge in my SDK PR today and propagate it to runtime tomorrow, this would be a no-concern.